tests/fixtures/mocks.py

```diff
@@ -0,0 +1,67 @@
+#  Copyright 2019 XLAB d.o.o.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from itertools import cycle
+
+import pytest
+from redfish_client import Connector
+
+
+def set_nested_key(obj, key, value):
+    initial_object = obj
+    for k in key[:-1]:
+        obj = obj.setdefault(k, {})
+    obj[key[-1]] = value
+    return initial_object
+
+
+def connector_mocker(mocker, method, is_ok=True, status_code=200):
+    mock_post = mocker.patch.object(Connector, method)
+    mock_post.return_value.ok = is_ok
+    mock_post.return_value.status_code = status_code
+    return mock_post
+
+
+@pytest.fixture
+def connector_post_mocker(mocker):
+    return connector_mocker(mocker, 'post')
+
+
+@pytest.fixture
+def connector_get_mocker(mocker):
+    return connector_mocker(mocker, 'get')
+
+
+@pytest.fixture
+def connector_get_system_mocker_with_side_effect(request, connector_get_mocker, mocker, system):
```

Comment:
Please rename to `cycle_values`, this is freaking too long name and it appears in tests all the time :) But kudos for such fixture

Reply:
Since we moved `system` fixture to a `SYSTEM_LENOVO` variable, maybe we could parametrize this fixture to also accept other dictionaries not just `SYSTEM_LENOVO`?